src/current/v23.1/live-migration-service.md

```diff
@@ -0,0 +1,646 @@
+| `--orchestrator-url` | The URL for the orchestrator, using the [configured port](#service-type). Prefix the URL with `https` instead of `http` when using [certificates](#security). This flag is required unless the value is exported as an environment variable using `export CLI_ORCHESTRATOR_URL="{orchestrator-URL}"`. |
+| `--tls-ca-cert`      | The path to the CA certificate. This can also be [exported](#configure-orchestrator-and-client-certificates) as an environment variable using `export CLI_TLS_CA_CERT="{path-to-cli-ca-cert}"`.                                                                                                       |
+| `--tls-client-cert`  | The path to the client certificate. This can also be [exported](#configure-orchestrator-and-client-certificates) as an environment variable using `export "{path-to-cli-client-cert}"`.                                                                                                               |
+| `--tls-client-key`   | The path to the client key. This can also be [exported](#configure-orchestrator-and-client-certificates) as an environment variable using `export "{path-to-cli-client-key}"`.                                                                                                                        |
+
+## Shadowing modes
+
+The LMS can be configured to shadow production traffic from the source database and validate the query results on the target. The exact behavior is configured with the [`shadowMode`](#shadowing) Helm value.
+
+### `none`
+
+<img src="{{ 'images/v23.2/migrations/lms_none.svg' | relative_url }}" alt="MOLT LMS shadowing mode - none" />
+
+`shadowMode: none` disables shadowing.
+
+- The LMS sends application requests to the source of truth only.
+- Query results from the source of truth are returned to the application.
+- Writes must be manually replicated from the source database to the target database.
+
+You can use this mode to perform a [consistent cutover](#consistent-cutover), along with another tool such as [change data capture (CDC)]({% link {{ page.version.version }}/cdc-queries.md %}) that replicates writes to the target database. {% comment %}For an example, see [Consistent cutover without shadowing](#consistent-cutover-without-shadowing).{% endcomment %}
+
+### `async`
+
+<img src="{{ 'images/v23.2/migrations/lms_async.svg' | relative_url }}" alt="MOLT LMS shadowing mode - async" />
+
+`shadowMode: async` writes to both databases.
+
+- The LMS sends application requests to the source of truth and target database in asynchronous threads, and waits only for the source of truth to respond.
+- Query results from the source of truth are returned to the application.
+- If an asynchronous request has not yet completed, subsequent asynchronous requests will be dropped.
```

Comment:
Yes. Those requests are dropped forever. I think it should be the right thing to do. If a few queries go really poorly on crdb and you create memory backpressure of storing a bunch of queries in memory, you have the chance to oomkill or even throttle your LMS container which is still handling your hot path traffic.

Reply:
To play a bit of devil's advocate, I'd have to imagine that a small amount of queuing will solve most of the incidents where there's unexpected latency (especially since things worked well at the SoT).  Would it not be better to start queueing and only drop things once the queue fills up?

Reply:
(Clarifying this to "permanently dropped").